### PR TITLE
Terminate microscope in CI properly

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4452)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/microscope.go
Comment (eloycoto) [inline test/k8sT/microscope.go:37]: AfterFailed function here.

File: test/helpers/kubectl.go
Comment (eloycoto) [inline test/helpers/kubectl.go:460]: return err  here.

**Comment by nebril** [comment]
fixes #4011

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Comment by nebril** [comment]
@eloycoto fixed

**Comment by nebril** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

File: test/helpers/kubectl.go
Comment (vadorovsky) [inline test/helpers/kubectl.go:433]: Just `kub.Apply(...)`? `_` variable isn't needed if you don't want to get any value.

File: test/helpers/ssh_command.go
Comment (vadorovsky) [inline test/helpers/ssh_command.go:219]: same here, in the whole function

File: test/helpers/kubectl.go
Comment (vadorovsky) [inline test/helpers/kubectl.go:435]:
```
if err := kub.WaiforPods(...); err != nil {
    return err, nil
}
```

File: test/k8sT/microscope.go
Comment (vadorovsky) [inline test/k8sT/microscope.go]: A bit easier approach would be

```
res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep microscope")
res.ExpectSuccess()
```

No need to look for the output.

**Review by vadorovsky — COMMENTED** [review]
small nits

File: test/helpers/kubectl.go
Comment (ianvernon) [reply test/helpers/kubectl.go:433]: @mrostecki in the CI code, we have hit unintentional bugs by ignoring return values of functions. I have requested in the past that we always specify that we are ignoring the result of a function with `_ =` so that it is clear that the writer of the test intended to ignore the result, and is not unintentionally doing so.

File: test/k8sT/microscope.go
Comment (ianvernon) [inline test/k8sT/microscope.go]: We still have the convention of `K8sValidated*` for our CI test names. Please change to `K8sValidatedMicroscope` so that the test framework will pick up the test when it is ran in CI builds.

**Review by ianvernon — DISMISSED** [review]
One small change needed. Thanks for adding the test!

File: test/helpers/kubectl.go
Comment (vadorovsky) [reply test/helpers/kubectl.go:433]: OK, thanks for explanation.

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/microscope.go
Comment (nebril) [reply test/k8sT/microscope.go]: Done

File: test/k8sT/microscope.go
Comment (nebril) [reply test/k8sT/microscope.go]: Thanks, I had to use `-f` flag for pgrep, also this should fail, since we don't want microscope to be running anymore. Good catch

**Comment by nebril** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Comment by vadorovsky** [comment]
test-me-please

**Review by eloycoto — DISMISSED** [review]

**Comment by nebril** [comment]
test-me-please

**Comment by eloycoto** [comment]
@nebril could you get the link of the failing build? Why it fails?

**Comment by vadorovsky** [comment]
Well, now it passed. It was some random VM provisioning error, it occurred two times in the row.

**Comment by eloycoto** [comment]
So, we have this fail: 

https://jenkins.cilium.io/view/PR/job/Cilium-PR-Ginkgo-Tests-Validated/4121/testReport/junit/k8s-1/7/K8sValidatedMicroscope_Runs_microscope/

Commit: 813e8e54a63f85b2d0ebe424944e6d61320141e8

And the current commit is the same: 813e8e54a63f85b2d0ebe424944e6d61320141e8

Logs: 
[f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip](https://github.com/cilium/cilium/files/2098197/f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip)

@nebril could you triage this one? I want to make sure that no other flake around microscope. 

And we have other fail: 
https://jenkins.cilium.io/view/PR/job/Cilium-PR-Ginkgo-Tests-Validated/4117/testReport/junit/k8s-1/10/K8sValidatedChaosTest_Endpoint_can_still_connect_while_Cilium_is_not_running/

That it's the provision one.

**Comment by ianvernon** [comment]
Holding off on re-reviewing due to the issue @eloycoto pointed out with https://jenkins.cilium.io/view/PR/job/Cilium-PR-Ginkgo-Tests-Validated/4121/testReport/junit/k8s-1/7/K8sValidatedMicroscope_Runs_microscope/ , as this appears to show that the microscope process is not being terminated in the pod.

File: test/helpers/ssh_command.go
Comment (ianvernon) [inline test/helpers/ssh_command.go:227]: I've been thinking about this, and worse comes to worse, if sending signals over the SSH session doesn't work, we could try to kill it by doing a `kubectl exec <kill microscope>` in the cilium pod? :/

Comment (nebril) [reply test/helpers/ssh_command.go:227]: Yeah, this is an ultimate fallback, but we would need to retrieve the pod and namespace from command, since this particular command is run in microscope pod.

**Comment by nebril** [comment]
test-me-please

**Comment by ianvernon** [comment]
@nebril I will run this locally, and if microscope processes are not running any more after running tests on the VMs, will merge.

**Review by ianvernon — APPROVED** [review]
Code looks good. For good measure, validated locally that there are no leftover python / microscope processes left around on the nodes:

```
$ ps aux | grep "[p]ython\|[m]icroscope"
```

**Review by eloycoto — APPROVED** [review]

**Pull request merged**
